storage-access-api/helpers.js

```diff
@@ -150,10 +150,32 @@ async function CanFrameWriteCookies(frame, keep_after_writing = false) {
   return can_write;
 }
 
+// Sets a cookie in an unpartitioned context by creating a new frame
+// and requesting storage access in the frame.
+async function SetUnpartitionedCookie(origin) {
+  let frame = await CreateFrame(`${origin}/storage-access-api/resources/script-with-cookie-header.py?script=embedded_responder.js`);
+  await SetPermissionInFrame(frame, [{ name: 'storage-access' }, 'granted']);
+  await RequestStorageAccessInFrame(frame);
+  await SetDocumentCookieFromFrame(frame, `cookie=unpartitioned;Secure;SameSite=None;Path=/`);
+  await SetPermissionInFrame(frame, [{ name: 'storage-access' }, 'prompt']);
```

Comment:
This line assumes that the permission state was `prompt` when this helper was called. Might be nice to make it just restore whatever the previous permission state was, instead? Then this helper would be a bit easier to use correctly, since it cleans up after itself more consistently.

Reply:
Good catch! I missed that. I'd like to do that actually.

The challenging part here is that we can't know if the `'storage-access'` permission was `"denied"`. We only have `navigator.permissions.get` to read permissions and we never return `"denied"` there. Given that, I would be willing to rename the function to indicate that it resets the permission as well.

Thoughts?

Reply:
That's true, we're kind of forced to use `"prompt"` instead of `"denied"`, at best. I like the idea of renaming the function to make it clear that it resets the permission - that way it'll be clear at callsites that they need to explicitly set the permission to whatever they need it to be after running this.

Reply:
I'll change the name of the function to `SetFirstPartyCookieAndUnsetStorageAccessPermission`.

At very least it will be verbose enough to give future developers pause :)